streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -77,23 +77,41 @@ public void testNewName() {
         assertEquals("Y-0000000001", builder.newName("Y-"));
         assertEquals("Z-0000000002", builder.newName("Z-"));
 
-        KStreamBuilder newBuilder = new KStreamBuilder();
+        final KStreamBuilder newBuilder = new KStreamBuilder();
 
         assertEquals("X-0000000000", newBuilder.newName("X-"));
         assertEquals("Y-0000000001", newBuilder.newName("Y-"));
         assertEquals("Z-0000000002", newBuilder.newName("Z-"));
     }
 
+
+    @Test
+    public void shouldHaveSinkTopicForStreamTo() {
+        final KStream<String, String> source = builder.stream("topic-source");
+        source.to("topic-sink");
+
+        final MockProcessorSupplier<String, String> processorSupplier = new MockProcessorSupplier<>();
+
+        source.process(processorSupplier);
+
+        driver = new KStreamTestDriver(builder);
+        driver.setTime(0L);
+
+        driver.process("topic-source", "A", "aa");
```

Comment:
All internal topics are prefixed by `application.id` and suffixed with `-repartioning`. Thus, in `StreamTestDriver$MockRecordCollector.send()` you could just check the topic name, and only call `process()` for "internal topics".

On the other hand, I am just thinking how this would affect calls to `through(...)` -- if we only forward for internal topics, we would break this part in the test driver. Thus, I now believe the current patch is actually correct as-is :) \cc @guozhangwang 

@wimvanleuven The test you have is good. With regard to `.through()` it might be worth adding a test for this case, too. Can be very similar to the test you added already: `builder.stream().through().process()` should cover that case. WDYT?